clang/lib/CodeGen/CGDebugInfo.cpp

```diff
@@ -3416,6 +3437,27 @@ llvm::DIMacroFile *CGDebugInfo::CreateTempMacroFile(llvm::DIMacroFile *Parent,
   return DBuilder.createTempMacroFile(Parent, Line, FName);
 }
 
+llvm::DILocation *CGDebugInfo::CreateTrapFailureMessageFor(
+    llvm::DebugLoc TrapLocation, StringRef Prefix, StringRef FailureMsg) {
+  // Create debug info that describes a fake function whose name is the failure
+  // message.
+  std::string FuncName(Prefix);
+  if (!FailureMsg.empty()) {
+    // A space in the function name identifies this as not being a real function
+    // because it's not a valid symbol name.
+    FuncName += ": ";
+    FuncName += FailureMsg;
+  }
+
+  assert(FuncName.size() > 0);
+  assert(FuncName.find(' ') != std::string::npos &&
```

Comment:
the pre condition (as stated in the function documentation) is that `Prefix` should have a space.
But we add a space whenever `FailureMsg` is not empty, so this assert is not catching a pre condition violation in all cases

Reply:
I think it means `Prefix` is required to have a space only when `FailureMsg` is empty. If `FailureMsg` is empty, `FuncName` is the same as `Prefix`, so the checks seem correct to me. @delcypher is that correct?

Reply:
We can better express this intent by moving the assert _before_ the `if` and by rewriting it as the faster `assert(!FailureMessage.empty() || FuncName.find(' ') != npos)`

Reply:
I think that should be fine. We should also make sure someone doesn't accidentally remove the space in `FuncName` inside the `if`, but I guess that's okay since there's a comment there.

Reply:
I think the right answer here is:
We guarantee not to collide with other symbols by use the reserved `__` prefix. Any consumer that wants to identify such a symbol checks for the `__llvm_builtin_trap` prefix to unambiguously identify these functions.

Reply:
The space is not needed for correctness.